src/devices/Mlx90640/Mlx90640.cs

```diff
@@ -0,0 +1,1072 @@
+                }
+                else
+                {
+                    pattern = chessPattern;
+                }
+
+                if (pattern == frameData[833])
+                {
+                    irData = frameData[pixelNumber];
+                    if (irData > 32767)
+                    {
+                        irData = irData - 65536;
+                    }
+
+                    irData = irData * gain;
+
+                    kta = _kta[pixelNumber] / ktaScale;
+                    kv = _kv[pixelNumber] / kvScale;
+                    irData = (float)(irData - _offset[pixelNumber] * (1 + kta * (ta - 25)) * (1 + kv * (vdd - 3.3)));
+
+                    if (mode != _calibrationModeEE)
+                    {
+                        irData = irData + _ilChessC[2] * (2 * ilPattern - 1) - _ilChessC[1] * conversionPattern;
+                    }
+
+                    irData = irData - _tgc * irDataCP[subPage];
+                    irData = irData / emissivity;
+
+                    alphaCompensated = (float)(SCALEALPHA * alphaScale / _alpha[pixelNumber]);
+                    alphaCompensated = alphaCompensated * (1 + _ksTa * (ta - 25));
+
+                    sx = alphaCompensated * alphaCompensated * alphaCompensated * (irData + alphaCompensated * taTr);
+                    sx = (float)(Math.Sqrt(Math.Sqrt(sx)) * _ksTo[1]);
+
+                    to = (float)(Math.Sqrt(Math.Sqrt(irData / (alphaCompensated * (1 - _ksTo[1] * 273.15) + sx) + taTr)) - 273.15);
+
+                    if (to < _ct[1])
+                    {
+                        range = 0;
+                    }
+                    else if (to < _ct[2])
+                    {
+                        range = 1;
+                    }
+                    else if (to < _ct[3])
+                    {
+                        range = 2;
+                    }
+                    else
+                    {
+                        range = 3;
+                    }
+
+                    to = (float)(Math.Sqrt(Math.Sqrt(irData / (alphaCompensated * alphaCorrR[range] * (1 + _ksTo[range] * (to - _ct[range]))) + taTr)) - 273.15);
+
+                    result[pixelNumber] = to;
+                }
+            }
+        }
+
+        private float GetTa(ushort[] frameData)
+        {
+            float ptat;
+            float ptatArt;
+            float vdd;
+            float ta;
+
+            vdd = GetVdd(frameData);
+
+            ptat = frameData[800];
+            if (ptat > 32767)
+            {
+                ptat = ptat - 65536;
+            }
+
+            ptatArt = frameData[768];
+            if (ptatArt > 32767)
+            {
+                ptatArt = ptatArt - 65536;
+            }
+
+            ptatArt = (float)((ptat / (ptat * _alphaPTAT + ptatArt)) * Math.Pow(2, 18));
+
+            ta = (float)((ptatArt / (1 + _kvPTAT * (vdd - 3.3)) - _vPTAT25));
+            ta = ta / _ktPTAT + 25;
+
+            return ta;
```

Comment:
```suggestion
            return Temperature.FromCelsius(ta);
```

assuming this is returning temperature in ceslius